tests/by-util/test_shuf.rs

```diff
@@ -147,6 +149,81 @@ fn test_head_count() {
     );
 }
 
+#[test]
+fn test_zero_head_count_pipe() {
+    let result = new_ucmd!().arg("-n0").pipe_in(vec![]).succeeds();
+    // Output must be completely empty, not even a newline!
+    result.no_output();
+}
+
+#[test]
+fn test_zero_head_count_pipe_explicit() {
+    let result = new_ucmd!().arg("-n0").arg("-").pipe_in(vec![]).succeeds();
+    result.no_stderr();
+
+    let result_seq: &str = result.stdout_str();
+    assert!(
+        result_seq.is_empty(),
+        "Output must be completely empty, not even newline"
+    );
+}
+
+#[test]
+fn test_zero_head_count_file_unreadable() {
+    let result = new_ucmd!()
+        .arg("-n0")
+        .arg("/invalid/unreadable")
+        .pipe_in(vec![])
+        .succeeds();
+    result.no_stderr();
+
+    let result_seq: &str = result.stdout_str();
+    assert!(
+        result_seq.is_empty(),
+        "Output must be completely empty, not even newline"
+    );
+}
+
+#[test]
+fn test_zero_head_count_file_touch_output() {
+    new_ucmd!()
+        .arg("-n0")
+        .arg("-o")
+        .arg("/invalid/unwritable")
+        .pipe_in(vec![])
+        .fails()
+        .stderr_contains("failed to open '/invalid/unwritable' for writing:");
+}
```

Comment:
There is a test function missing similar to this function to ensure the success case is working (i.e. the specified file is created).

Reply:
You're right, thanks!